src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -186,6 +186,11 @@ class full_thread_state {
   int num_times_retry_throw = 0;
   int num_times_split_retry_throw = 0;
   long time_blocked_nanos = 0;
+  long time_lost_nanos = 0;
+  long time_retry_running_nanos = 0;
+  std::chrono::time_point<std::chrono::steady_clock> retry_start_or_block_end;
```

Comment:
could there be a comment for these times on what they mean? I think it would improve on figuring out how the code should be used.